stream-ingest/src/main/java/com/dellemc/oe/ingest/EventWithTimestampWriter.java

```diff
@@ -0,0 +1,105 @@
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+
+/**
+ * A simple example app that uses a Pravega Writer to write to a given scope and stream.
+ */
+public class EventWithTimestampWriter {
+    // Logger initialization
+    private static final Logger LOG = LoggerFactory.getLogger(JSONWriter.class);
+
+    public final String scope;
+    public final String streamName;
+    public final URI controllerURI;
+
+    public EventWithTimestampWriter(String scope, String streamName, URI controllerURI) {
+        this.scope = scope;
+        this.streamName = streamName;
+        this.controllerURI = controllerURI;
+    }
+
+    // Create a JSON data for testing purpose
+    public static ObjectNode createJSONData() {
+        ObjectNode message = null;
+        try {
+            String data = "{\"sensorid\":" + Math.random() + ",\"time\":" + System.currentTimeMillis() + ",\"value\":" + Math.random();
+            // Deserialize the JSON message.
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode jsonNode = objectMapper.readTree(data);
+            message = (ObjectNode) jsonNode;
+            LOG.info("@@@@@@@@@@@@@ DATA >>>  " + message.toString());
+            return message;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void main(String[] args) {
+        final String scope = CommonParams.getScope();
+        final String streamName = CommonParams.getStreamName();
+        final String routingKey = CommonParams.getRoutingKeyAttributeName();
+        final URI controllerURI = CommonParams.getControllerURI();
+        JSONWriter ew = new JSONWriter(scope, streamName, controllerURI);
+        ew.run(routingKey);
+    }
+
+    public void run(String routingKey) {
+
+        try {
+            String streamName = "json-stream";
+            // Create client config
+            ClientConfig clientConfig = ClientConfig.builder()
+                    .controllerURI(controllerURI).build();
+            StreamManager streamManager = StreamManager.create(clientConfig);
+            StreamConfiguration streamConfig = StreamConfiguration.builder().build();
+            if (CommonParams.isPravegaStandalone()) {
+                streamManager.createScope(scope);
+            }
+            streamManager.createStream(scope, streamName, streamConfig);
+            // Create EventStreamClientFactory
+            EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scope, clientConfig);
+
+            // Create event writer
+            EventStreamWriter<JsonNode> writer = clientFactory.createEventWriter(
+                    streamName,
+                    new JsonNodeSerializer(),
+                    EventWriterConfig.builder().build());
+            // same data write every 1 sec
+            while (true) {
+                ObjectNode data = createJSONData();
+                writer.writeEvent(routingKey, data);
```

Comment:
Determine routingKey by looking in data object.

Reply:
Do not ignore the returned future. You should `get` it to make it clear that this is an async function.